lib/src/main/java/com/orange/ods/compose/component/card/OdsHorizontalCard.kt

```diff
@@ -88,76 +88,129 @@ fun OdsHorizontalCard(
     onButton1Click: (() -> Unit)? = null,
     onButton2Click: (() -> Unit)? = null
 ) {
-    val imageComposable: @Composable () -> Unit = {
-        HorizontalCardImage(
-            image = image,
-            contentScale = imageContentScale,
-            alignment = imageAlignment,
-            contentDescription = imageContentDescription,
-            backgroundColor = imageBackgroundColor
-        )
-    }
-
     OdsCard(
         modifier = modifier.fillMaxWidth(),
         onClick = onCardClick
     ) {
-        Column {
-            Row(modifier = Modifier.height(IntrinsicSize.Min)) {
-                if (imagePosition == OdsHorizontalCardImagePosition.Start) {
-                    imageComposable()
-                }
+        ConstraintLayout {
+            val (
+                imageRef,
+                titleRef,
+                subtitleRef,
+                textRef,
+                chainBottomSpacerRef, // A 0 dp spacer located at the bottom of the vertical chain composed of title, subtitle and text. Without this spacer, when text is gone, bottom margin of the chain is not respected
+                dividerRef,
+                button1Ref,
+                button2Ref
+            ) = createRefs()
 
-                Column(
-                    modifier = Modifier
-                        .weight(1f)
-                        .padding(dimensionResource(id = R.dimen.spacing_m))
-                ) {
-                    OdsTextH6(text = title)
-                    subtitle?.let {
-                        OdsTextSubtitle2(text = it)
-                    }
-                    text?.let {
-                        Text(
-                            modifier = Modifier.padding(
-                                top = dimensionResource(id = R.dimen.spacing_s)
-                            ),
-                            text = it,
-                            style = OdsTheme.typography.body1,
-                            maxLines = if (subtitle == null) 3 else 2,
-                            overflow = TextOverflow.Ellipsis
-                        )
+            // Divider is not always visible thus we need to add this barrier otherwise margin between divider and text vertical chain is not respected
+            val dividerTopBarrier = createTopBarrier(dividerRef)
+            val buttonsTopBarrier = createTopBarrier(button1Ref, button2Ref)
```

Comment:
I think we can simplify the ConstraintLayout and remove the barriers, if you can try that on your side: 

```kotlin
  ConstraintLayout {
      val (
          imageRef,
          titleRef,
          subtitleRef,
          textRef,
          dividerRef,
          button1Ref,
          button2Ref
      ) = createRefs()

      val imageSize = dimensionResource(R.dimen.card_horizontal_image_size)
      val smallSpacing = dimensionResource(id = R.dimen.spacing_s)
      val mediumSpacing = dimensionResource(id = R.dimen.spacing_m)

      Image(
          painter = image,
          contentDescription = imageContentDescription,
          contentScale = imageContentScale,
          modifier = Modifier
              .let { if (imageBackgroundColor != null) it.background(backgroundColor) else it }
              .constrainAs(imageRef) {
                  top.linkTo(parent.top)
                  bottom.linkTo(dividerRef.top)
                  when (imagePosition) {
                      OdsHorizontalCardImagePosition.Start -> start.linkTo(parent.start)
                      OdsHorizontalCardImagePosition.End -> end.linkTo(parent.end)
                  }
                  width = Dimension.value(imageSize)
                  height = Dimension.fillToConstraints.atLeast(imageSize)
              },
          alignment = imageAlignment
      )

      val chainRef = createVerticalChain(titleRef, subtitleRef, textRef, chainStyle = ChainStyle.Packed)
      constrain(chainRef) {
          top.linkTo(parent.top, mediumSpacing)
          bottom.linkTo(imageRef.bottom, mediumSpacing)
      }

      OdsTextH6(
          text = title,
          modifier = Modifier.constrainAs(titleRef) {
              when (imagePosition) {
                  OdsHorizontalCardImagePosition.Start -> {
                      start.linkTo(imageRef.end, margin = mediumSpacing)
                      end.linkTo(parent.end, margin = mediumSpacing)
                  }
                  OdsHorizontalCardImagePosition.End -> {
                      start.linkTo(parent.start, margin = mediumSpacing)
                      end.linkTo(imageRef.start, margin = mediumSpacing)
                  }
              }
              width = Dimension.fillToConstraints
          }
      )

      OdsTextSubtitle2(
          text = subtitle.orEmpty(),
          modifier = Modifier.constrainAs(subtitleRef) {
              start.linkTo(titleRef.start)
              end.linkTo(titleRef.end)
              width = Dimension.fillToConstraints
              visibility = if (subtitle != null) Visibility.Visible else Visibility.Gone
          }
      )

      Text(
          modifier = Modifier
              .padding(top = smallSpacing) // For some reason, margins inside a chain are not applied, a workaround is to apply padding before the constraints
              .constrainAs(textRef) {
                  start.linkTo(titleRef.start)
                  end.linkTo(titleRef.end)
                  width = Dimension.fillToConstraints
                  visibility = if (text != null) Visibility.Visible else Visibility.Gone
              },
          text = text.orEmpty(),
          style = OdsTheme.typography.body1,
          maxLines = if (subtitle == null) 3 else 2,
          overflow = TextOverflow.Ellipsis
      )

      OdsDivider(
          modifier = Modifier.constrainAs(dividerRef) {
              top.linkTo(imageRef.bottom)
              start.linkTo(parent.start)
              end.linkTo(parent.end)
              visibility = if (dividerEnabled && (button1Text != null || button2Text != null)) Visibility.Visible else Visibility.Gone
          }
      )

      OdsTextButton(
          modifier = Modifier.constrainAs(button1Ref) {
              top.linkTo(dividerRef.bottom)
              start.linkTo(parent.start, margin = smallSpacing)
              visibility = if (button1Text != null) Visibility.Visible else Visibility.Gone
          },
          text = button1Text.orEmpty(),
          onClick = { onButton1Click?.invoke() },
          style = OdsTextButtonStyle.Primary
      )

      OdsTextButton(
          modifier = Modifier.constrainAs(button2Ref) {
              top.linkTo(dividerRef.bottom)
              start.linkTo(button1Ref.end, margin = smallSpacing, goneMargin = smallSpacing)
              visibility = if (button2Text != null) Visibility.Visible else Visibility.Gone
          },
          text = button2Text.orEmpty(),
          onClick = { onButton2Click?.invoke() },
          style = OdsTextButtonStyle.Primary
      )
  }
```